src/Sylius/Bundle/AdminBundle/Storage/FilterStorage.php

```diff
@@ -13,26 +13,32 @@
 
 namespace Sylius\Bundle\AdminBundle\Storage;
 
+use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\HttpFoundation\Session\SessionInterface;
 
 final class FilterStorage implements FilterStorageInterface
 {
-    public function __construct(private SessionInterface $session)
+    public function __construct(private RequestStack $requestStack)
```

Comment:
Here and in other places we should allow passing two types `SessionInterface|RequestStack` and deprecate passing SessionInterface directly - like here: https://github.com/Sylius/Sylius/blob/symfony-6/src/Sylius/Bundle/UserBundle/EventListener/UserDeleteListener.php#L27